src/execution/syscall_handler_utils.rs

```diff
@@ -300,54 +222,83 @@ pub fn write_segment(vm: &mut VirtualMachine, ptr: &mut Relocatable, segment: Re
     Ok(())
 }
 
-impl SyscallResponse for SingleSegmentResponse {
-    fn write(self, vm: &mut VirtualMachine, ptr: &mut Relocatable) -> WriteResponseResult {
-        write_segment(vm, ptr, self.segment)
-    }
+pub trait SyscallHandler {
+    type Request;
+    type Response;
+    fn read_request(vm: &VirtualMachine, ptr: &mut Relocatable) -> SyscallResult<Self::Request>;
```

Comment:
This is already an improvement, but I think we could also generate the `read_request()` and `write_response()` code. WDYT?

Reply:
you mean noop default implementation?